android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRegistry.java

```diff
@@ -66,7 +67,12 @@ private synchronized void detachHandler(GestureHandler handler) {
       // Handler is in "prepared" state which means it is registered in the orchestrator and can
       // receive touch events. This means that before we remove it from the registry we need to
       // "cancel" it so that orchestrator does no longer keep a reference to it.
-      handler.cancel();
+      UiThreadUtil.runOnUiThread(new Runnable() {
```

Comment:
Should I extract this Runnable to the member variable?

Reply:
If it is created frequently then it may make sense. I don't think it is though, because it only happens when you change enabled property while there is an ongoing gesture

Reply:
This is called every time when you update/unmount GH which has associated view, so it may be worth doing.

Reply:
Actually, we need to capture `handler` variable and AFAIK there's no simple solution to reuse one Runnable to do that.